src/Dialog/Dialog.js

```diff
@@ -184,6 +184,7 @@ module.exports = kind(
 	*/
 	titleChanged: function () {
 		var title = this.getTitle();
+		this.$.title.set('accessibilityLabel', title);
 		this.$.title.setContent( this.get('uppercase') ? util.toUpperCase(title) : title );
 	},
 
```

Comment:
how about adding this feature as ariaObservers like others?
ex)
ariaObservers: [
    {path: ['title'], method: function () {
        var content = this.getTitle();
        this.setAriaAttribute('aria-label', content);
    }}
]